File: python/matrix/matrix.py
import sys
sys.path.append("numpy_path")
import numpy as np

class Matrix:
    def __init__(self, matrix_string):
        self.matrix = [[int(j) for j in i.split()] for i in matrix_string.splitlines()]
    
    def row(self, index):
        return self.matrix[index-1]

    def column(self, index):
        column = []
        [[column.append(self.matrix[i][index-1]) ] for i in range(len(self.matrix))]
        
        return column

File: javascript/matrix/matrix.js
import { transpose } from 'mathjs'
const parseRow = row => row.split(' ').map(Number);
const parseRows = rows => rows.split("\n").map(parseRow);

const parseColumn = column => column.split("\n").map(parseRow);



export class Matrix {
  constructor(raw) { this.raw = raw; }

  get rows() {
    return this._rows_ = this._rows_ ||  parseRows(this.raw);
  }

  get columns() {
    return  this._columns_ = this._columns_ || transpose(this.rows);
  }
}
